test/framework/src/main/java/org/opensearch/test/OpenSearchIntegTestCase.java

```diff
@@ -2349,4 +2354,83 @@ protected ClusterState getClusterState() {
         return client(internalCluster().getClusterManagerName()).admin().cluster().prepareState().get().getState();
     }
 
+    /**
+     * Refreshes the indices in the cluster and checks if active/started replica shards
+     * caught up with primary shard when Segment Replication is enabled.
+     * This doesn't wait for inactive/non-started replica shards to become active/started.
+     */
+    protected void waitForReplication(String... indices) {
+        refresh(indices);
+        if (indices.length == 0) {
+            indices = getClusterState().routingTable().indicesRouting().keySet().toArray(String[]::new);
+        }
+        try {
+            for (String index : indices) {
+                if (isInternalCluster() && isSegmentReplicationEnabledForIndex(index)) {
+                    IndexRoutingTable indexRoutingTable = getClusterState().routingTable().index(index);
+                    if (indexRoutingTable != null) {
+                        assertBusy(() -> {
+                            for (IndexShardRoutingTable shardRoutingTable : indexRoutingTable) {
+                                final ShardRouting primaryRouting = shardRoutingTable.primaryShard();
+                                if (primaryRouting.state().toString().equals("STARTED")) {
+                                    if (isSegmentReplicationEnabledForIndex(index)) {
+                                        final List<ShardRouting> replicaRouting = shardRoutingTable.replicaShards();
+                                        final IndexShard primaryShard = getIndexShard(primaryRouting, index);
+                                        for (ShardRouting replica : replicaRouting) {
+                                            if (replica.state().toString().equals("STARTED")) {
+                                                IndexShard replicaShard = getIndexShard(replica, index);
+                                                assertEquals(
+                                                    "replica shards haven't caught up with primary",
+                                                    getLatestSegmentInfoVersion(primaryShard),
+                                                    getLatestSegmentInfoVersion(replicaShard)
+                                                );
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        }, 30, TimeUnit.SECONDS);
+                    }
+                } else if (isInternalCluster() == false && isSegmentReplicationEnabledForIndex(index)) {
```

Comment:
My truth condition if `(isInternalCluster() && isSegmentReplicationEnabledForIndex(index))` allows waiting for replication only when cluster is internal and segrep enabled index. 

This else if condition `else if (isInternalCluster() == false && isSegmentReplicationEnabledForIndex(index))` condition is required because we only want to throw exception when cluster is external and segrep enabled index.

For all other case like
->  when cluster is external and docrep enabled index, we should skip waitForReplication and should not throw exception.
-> When cluster is internal and docrep enabled index, we should skip waitForReplication and should not throw exception.

Without using `else if (isInternalCluster() == false && isSegmentReplicationEnabledForIndex(index))` and simply using else, we will not cover for other cases mentioned above and throw exception incorrectly.

Reply:
Done, will update in next commit